File: problemas/iniciante/1097/1097.exs
defmodule Modulo do
  def imprime(i, j) do
    IO.puts "I=#{i} J=#{j}"
    IO.puts "I=#{i} J=#{j - 1}"
    IO.puts "I=#{i} J=#{j - 2}"
    if i == 9 do
      :ok
    else
      imprime(i + 2, j + 2)
    end
  end
end

Modulo.imprime(1, 7)

File: problemas/iniciante/1097/1097.hs
import Text.Printf

imprime :: Int -> Int -> IO ()
imprime 9 j = do
  printf "I=9 J=%d\n" j
  printf "I=9 J=%d\n" (j - 1)
  printf "I=9 J=%d\n" (j - 2)
imprime i j = do
  printf "I=%d J=%d\n" i j
  printf "I=%d J=%d\n" i (j - 1)
  printf "I=%d J=%d\n" i (j - 2)
  imprime (i + 2) (j + 2)

main :: IO ()
main = imprime 1 7